dask/dataframe/core.py

```diff
@@ -4555,14 +4556,80 @@ def sort_values(
             **kwargs,
         )
 
+    def _check_index(self, other: Any) -> Optional[Any]:
+        """Internal method to check the column passed to set_index.
+
+        Returns None if this column is already the index.
+        """
+
+        # Check other can be translated to column name or column object
+        index = other
+        if not isinstance(index, str):
+            if isinstance(index, Sequence):
+                # Accept ["a"], but not [["a"]]
+                if len(index) == 1 and (
+                    isinstance(index[0], str) or not isinstance(index[0], Sequence)
+                ):
+                    index = index[0]
+                else:
+                    raise NotImplementedError(
+                        "Dask dataframe does not yet support multi-indexes.\n"
+                        f"You tried to index with this index: {index}\n"
+                        "Indexes must be single columns only."
+                    )
+
+            # Can also be a frame
+            elif isinstance(index, DataFrame):
+                raise NotImplementedError(
+                    "Dask dataframe does not yet support multi-indexes.\n"
+                    f"You tried to index with several columns: {list(index.columns)}\n"
+                    "Indexes must be single columns only."
+                )
+
+        # Now index should be either a series...
+        if isinstance(index, Series):
+            # If it's the current index, nothing to do
+            if index._name == self.index._name:
+                return None
+            # Otherwise, should have same length than the index
```

Comment:
```suggestion
            # Otherwise, should have same length as the existing index
```

Reply:
Agreed

Reply:
Finally I've removed the length checking, so this doesn't apply anymore.